code/game/objects/items/radio/radio.dm

```diff
@@ -292,10 +292,22 @@
 		return
 
 	var/area/A = get_area(src)
+	var/radio_disruption = CAVE_NO_INTERFERENCE
 	if(!isnull(A) && (A.ceiling >= CEILING_UNDERGROUND) && !(A.flags_area & ALWAYS_RADIO))
+		radio_disruption = CAVE_MINOR_INTERFERENCE
 		if(A.ceiling >= CEILING_DEEP_UNDERGROUND)
+			radio_disruption = CAVE_FULL_INTERFERENCE
+
+	var/list/inplace_interference = list(radio_disruption)
+	SEND_SIGNAL(talking_movable, COMSIG_CAVE_INTERFERENCE_CHECK, inplace_interference)
+	radio_disruption = inplace_interference[1]
+
+	switch(radio_disruption)
+		if(CAVE_MINOR_INTERFERENCE)
+			signal.data["compression"] += rand(20, 40)
+		if(CAVE_FULL_INTERFERENCE)
 			return
-		signal.data["compression"] += rand(20, 40)
+		else
```

Comment:
```suggestion
```

Reply:
Personal style of mine to have an empty else in switch cases where one or more potential cases are not caught in the switch's ifs, can remove it though if disliked.